src/main/java/org/dita/dost/writer/AbstractChunkTopicParser.java

```diff
@@ -242,12 +242,12 @@ FileInfo generateFileInfo(final URI output) {
     URI generateOutputFilename(final String id) {
         final FileInfo cfi = job.getFileInfo(currentFile);
         URI result = cfi.result.resolve(id + FILE_EXTENSION_DITA);
-        URI temp = tempFileNameScheme.generateTempFileName(result);
```

Comment:
I think this will break things. The `generateTempFileName` method will turn a source file name into a temporary file name, e.g. a hash of the absolute URI. Just taking the source file name and using that is probably incorrect.

Reply:
I thought so initially ... but in the course of changing this line, I added 30 or 40 debug lines, and that doesn't seem to be the case with the current code (you're commenting on the old code but I assume you're questioning the change away from this). Details:

- `currentFile` is the map
- `URI result` creates a URI using the ID of the new chunked topic; it creates the file name as a "source file", in the same source directory as `currentFile'. So you get a value like: `/path/to/map/topicid.dita`
- `generateTempFilename` returns the portion of that path that would exist inside `temp` -- the portion relative to the common directory accounting for uplevels. If there are no uplevels, it just returns `topicid.dita`. If uplevels goes up one directory, it returns `map/topicid.dita`. 
- That portion of the path is then evaluated, and the file generated in temp, _relative to the chunked document_. So with uplevels, when chunking `temp/map/doc.dita`, we try to chunk the new topic as `temp/map/map/topicid.dita` (This fails further down, meaning something isn't added to a stack, which even later results in an NPE.)
- The method name is confusing, but ... `generateTempFileName` isn't generating a hash, it's taking the already-determined file name in a source directory, and returning the portion of path which fits in `temp`.

So there is confusion between `generateTempFilename(result)` -- which returns part of the URI without modifying it -- and `generateFilename()`, which actually generates a name.

If you can ignore the method names, what's happening today is:

- Generate a file name with the topic ID + `.dita`, as if it was a source file
- Take from that the portion that fits within temp -- _this bit is broken and later ends up with an NPE_
- If we couldn't get a file name, OR if that file name already exists within temp, go into the exception code on line 246 to find a new file name
- Exception code calls `generateFilename()` to get a hash. Today, that code also has the same problem as above (hash is created as a source file, we take the part that fits within `temp`, and then evaluate that relative to the chunked doc). So I've made the same fix within the exception code -- the hashed name should be relative to the chunked doc, regardless of uplevels.